testing: shadow-newpid.patch fails to build on Fedora 25

As seen when running #624 :
```
Using cache at /home/vagrant/.kpatch/src
Testing patch file
checking file fs/proc/array.c
checking file kernel/exit.c
checking file kernel/fork.c
Reading special section data
Building original kernel
Building patched kernel
ERROR: kpatch build failed. Check /home/vagrant/.kpatch/build.log for more details.
```
```
...
+ exec ld -m elf_x86_64 --emit-relocs --build-id -X -o .tmp_vmlinux1 -T ./arch/x86/kernel/vmlinux.lds arch/x86/kernel/head_64.o arch/x86/kernel/head64.o arch/x86/kernel/ebda.o arch/x86/kernel/platform-quirks.o init/built-in.o --start-group usr/built-in.o arch/x86/built-in.o kernel/built-in.o certs/built-in.o mm/built-in.o fs/built-in.o ipc/built-in.o security/built-in.o crypto/built-in.o block/built-in.o lib/lib.a arch/x86/lib/lib.a lib/built-in.o arch/x86/lib/built-in.o drivers/built-in.o sound/built-in.o firmware/built-in.o arch/x86/pci/built-in.o arch/x86/power/built-in.o arch/x86/video/built-in.o arch/x86/ras/built-in.o net/built-in.o virt/built-in.o --end-group
kernel/built-in.o: In function `_do_fork':
/home/vagrant/.kpatch/src/kernel/fork.c:1794: undefined reference to `kpatch_shadow_alloc'
kernel/built-in.o: In function `do_exit':
/home/vagrant/.kpatch/src/kernel/exit.c:832: undefined reference to `kpatch_shadow_free'
fs/built-in.o: In function `task_context_switch_counts':
/home/vagrant/.kpatch/src/fs/proc/array.c:366: undefined reference to `kpatch_shadow_get'
/home/vagrant/.kpatch/src/Makefile:953: recipe for target 'vmlinux' failed
make[1]: *** [vmlinux] Error 1
make[1]: Leaving directory '/home/vagrant/.kpatch/obj'
Makefile:150: recipe for target 'sub-make' failed
make: *** [sub-make] Error 2
```

**OS:** Fedora 25

**kernel version**: 4.8.6-300.fc25.x86_64
**gcc version**: gcc (GCC) 6.2.1 20160916 (Red Hat 6.2.1-2)

--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -513,10 +513,7 @@ patch -N -p1 < "$APPLIEDPATCHFILE" >> "$LOGFILE" 2>&1 || die
 mkdir -p "$TEMPDIR/orig" "$TEMPDIR/patched"
 KPATCH_GCC_TEMPDIR=$TEMPDIR
 export KPATCH_GCC_TEMPDIR
-# TODO: remove custom LDFLAGS and ugly "undefined reference" grep when core
-# module gets moved to the kernel tree
 CROSS_COMPILE="$TOOLSDIR/kpatch-gcc " \
-	LDFLAGS_vmlinux="--warn-unresolved-symbols" \
 	KBUILD_MODPOST_WARN=1 \
 	make "-j$CPUS" $TARGETS "O=$OBJDIR"  >> "$LOGFILE" 2>&1 || die
 [[ "${PIPESTATUS[0]}" -eq 0 ]] || die

--- a/kpatch-build/kpatch-gcc
+++ b/kpatch-build/kpatch-gcc
@@ -58,6 +58,10 @@ elif [[ "$TOOLCHAINCMD" = "ld" ]] ; then
 					cp -f "$obj" "$KPATCH_GCC_TEMPDIR/module/$obj"
 					break
 					;;
+				.tmp_vmlinux*|vmlinux)
+					args+=(--warn-unresolved-symbols)
+					break
+					;;
 				*)
 					break
 					;;

